tests/test_visualize.py

```diff
@@ -21,6 +21,19 @@ def test_get_flattened_orientation(self):
         R = np.array(_get_flattened_orientation(R, 1)).reshape(4, 4)
         self.assertAlmostEqual(np.linalg.det(R), 1)
 
+    def test_get_frame(self):
+        frame = _get_frame(np.eye(3))
+        self.assertLessEqual(
+            np.unique(frame.reshape(-1, 6), axis=0, return_counts=True)[1].max(),
+            1
+        )
+        dx, counts = np.unique(
+            np.diff(frame, axis=-2).squeeze().astype(int), axis=0, return_counts=True
+        )
+        self.assertEqual(dx.ptp(), 1)
+        self.assertEqual(counts.min(), 4)
+        self.assertEqual(counts.max(), 4)
```

Comment:
I don't know what any of these are checking, nor why they should return the integers they do. They don't necessarily need any change, but please specify `msg=` to clarify.